sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/CalciteToDBSPCompiler.java

```diff
@@ -3363,14 +3378,14 @@ private DBSPNode compileCreateView(CreateViewStatement view) {
             this.addOperator(vo);
             o = new DBSPSinkOperator(
                     view.getCalciteObject(), view.relationName,
-                    view.getStatement(), struct, meta, vo.outputPort());
+                    view.getStatement(), declaredStruct, meta, vo.outputPort());
```

Comment:
i haven't looked at the code closely but does this need to fall back to `struct` when `declare`/`declaredStruct` is `null`?

Reply:
Yes, if there is no view declaration declareStruct is set to struct in line 3354